src/widgets/free-text-input.jsx

```diff
@@ -0,0 +1,580 @@
+};
+
+var FreeTextInput = React.createClass({
+    propTypes: {
+        currentValue: React.PropTypes.string,
+        enabledFeatures: EnabledFeatures.propTypes,
+        apiOptions: ApiOptions.propTypes,
+        coefficient: React.PropTypes.bool,
+        answerForms: React.PropTypes.arrayOf(React.PropTypes.shape({
+            name: React.PropTypes.string.isRequired
+        })),
+        labelText: React.PropTypes.string,
+        reviewModeRubric: React.PropTypes.object,
+    },
+
+    getDefaultProps: function() {
+        return {
+            currentValue: "",
+            enabledFeatures: EnabledFeatures.defaults,
+            apiOptions: ApiOptions.defaults,
+            coefficient: false,
+            answerForms: [],
+            labelText: ""
+        };
+    },
+
+    render: function() {
+
+        var rubric = this.props.reviewModeRubric;
+        if (rubric) {
+            var score = this.simpleValidate(rubric);
+            var correct = score.type === "points" &&
+                          score.earned === score.total;
+
+            var answerBlurb = null;
+            if (!correct) {
+                var correctAnswers = _.filter(
+                    rubric.answers, (answer) => answer.status === "correct");
+                var answerComponents = _.map(correctAnswers, (answer, key) => {
+                    // Figure out how this answer is supposed to be displayed
+                    if (answer.answerForms && answer.answerForms[0]) {
+                        // NOTE(johnsullivan): This isn't exactly ideal, but
+                        // it does behave well for all the currently known
+                        // problems. See D14742 for some discussion on
+                        // alternate strategies.
+                        format = answer.answerForms[0]
+                    }
+
+                    var answerString = answer.value;
+                    return <span key={key} className="perseus-possible-answer">
+                        {answerString}
+                    </span>
+                });
+                answerBlurb = <span className="perseus-possible-answers">
+                    {answerComponents}
+                </span>;
+            }
+        }
+
+        var classes = {};
+        classes[ApiClassNames.CORRECT] =
+            rubric && correct && this.props.currentValue;
+        classes[ApiClassNames.INCORRECT] =
+            rubric && !correct && this.props.currentValue;
+        classes[ApiClassNames.UNANSWERED] = rubric && !this.props.currentValue;
+
+        var input = <TextArea
+            ref="input"
+            value={this.props.currentValue}
+            onChange={this.handleChange}
+            className={classNames(classes)}
+            type={this._getInputType()}
+            examples={this.examples()}
+            shouldShowExamples={this.shouldShowExamples()}
+            onFocus={this._handleFocus}
+            onBlur={this._handleBlur}
+            {...this.props} />;
```

Comment:
Damn, need to work out how to add in the object properties properly into TextArea, any ideas?